Mathlib/Tactic/Cases.lean

```diff
@@ -101,6 +123,28 @@ elab (name := induction') "induction' " tgts:(Parser.Tactic.casesTarget,+)
           (generalized := fvarIds) (toClear := targetFVarIds) (toTag := toTag)
         setGoals <| (subgoals ++ result.others).toList ++ gs
 
+/-- The `cases'` tactic is similar to the `cases` tactic in Lean 4 core, but the syntax for giving
+names is different:
+
+```
+example (h : p ∨ q) : q ∨ p := by
+  cases h with
+  | inl hp => exact Or.inr hp
+  | inr hq => exact Or.inl hq
+
+example (h : p ∨ q) : q ∨ p := by
+  cases' h with hp hq
+  · exact Or.inr hp
+  · exact Or.inl hq
+
+example (h : p ∨ q) : q ∨ p := by
+  rcases h with hp | hq
+  · exact Or.inr hp
+  · exact Or.inl hq
+```
+
+Prefer `cases` or `rcases` when possible, because these tactics promote structured proofs.
```

Comment:
There is [rcases vs obtain](https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/Is.20cases'.20deprecated.3F/near/433757205), so I wonder if `rcases` shouldn't be mentioned (in favour of `obtain`). But there seems to be consent there to leave that for a future refactor tool...